ChangeLog.d/mbedtls_pk_sign_ext.txt

```diff
@@ -0,0 +1,2 @@
+Features
+   * Add accessor to sign with different key type in key context.
```

Comment:
I forgot: when we introduce a new function, the ChangeLog entry should mention the name of the new function, so that interested users know where to look. Suggestion: "Add mbedtls_pk_sign_ext() which allows generating RSA-PSS signatures when PSA Crypto is enabled."